addons/velero/1.6.2/install.sh

```diff
@@ -162,13 +227,142 @@ function velero_patch_http_proxy() {
+    if [ ! "$KOTSADM_DISABLE_S3" == 1 ] || [ -n "$ROOK_VERSION" ] || [ -n "$MINIO_VERSION" ]; then 
+        echo ****EXIT*****
+        return
+    fi
+    echo ****CONTINUE*****
+
+    # if an object store isn't installed don't migrate
+    # TODO (dans): this doeesn't support minio in a non-standard namespace
+    if (! kubernetes_resource_exists rook-ceph deployment rook-ceph-rgw-rook-ceph-store-a) && (! kubernetes_resource_exists minio deployment minio); then 
+        return
+    fi
+
+    echo ****WOW*****
+
+    export VELERO_S3_HOST=
+    export VELERO_S3_ACCESS_KEY_ID=
+    export VELERO_S3_ACCESS_KEY_SECRET=
+    if kubernetes_resource_exists rook-ceph deployment rook-ceph-rgw-rook-ceph-store-a; then 
+        echo "Previous installation of Rook Ceph detected."
+        VELERO_S3_HOST="rook-ceph-rgw-rook-ceph-store.rook-ceph"
+        VELERO_S3_ACCESS_KEY_ID=$(kubectl -n rook-ceph get secret rook-ceph-object-user-rook-ceph-store-kurl -o yaml | grep AccessKey | head -1 | awk '{print $2}' | base64 --decode)
+        VELERO_S3_ACCESS_KEY_SECRET=$(kubectl -n rook-ceph get secret rook-ceph-object-user-rook-ceph-store-kurl -o yaml | grep SecretKey | head -1 | awk '{print $2}' | base64 --decode)
+    else 
+        echo "Previous installation of Minio detected."
+        VELERO_S3_HOST="minio.minio"
+        VELERO_S3_ACCESS_KEY_ID=$(kubectl -n minio get secret minio-credentials -ojsonpath='{ .data.MINIO_ACCESS_KEY }' | base64 --decode)
+        VELERO_S3_ACCESS_KEY_SECRET=$(kubectl -n minio get secret minio-credentials -ojsonpath='{ .data.MINIO_SECRET_KEY }' | base64 --decode)
+    fi
+
+    # If this is run through `velero_already_applied`, we need to create base kustomization file
+    if [ ! -f "$dst/kustomization.yaml" ];then
+        render_yaml_file "$src/tmpl-kustomization.yaml" > "$dst/kustomization.yaml"
+    fi
+
+    # TODO (dans): figure out if there is enough space create a new volume with all the snapshot data
+
+    # create secret for migration init container to pull from object store
+    render_yaml_file "$src/tmpl-s3-migration-secret.yaml" > "$dst/s3-migration-secret.yaml"
+    insert_resources "$dst/kustomization.yaml" s3-migration-secret.yaml
+
+    # create configmap that holds the migration script
+    render_yaml_file "$src/tmpl-s3-migration-configmap.yaml" > "$dst/s3-migration-configmap.yaml"
+    insert_resources "$dst/kustomization.yaml" s3-migration-configmap.yaml
+
+    # add patch to add init container for migration
+    render_yaml_file "$src/tmpl-s3-migration-deployment-patch.yaml" > "$dst/s3-migration-deployment-patch.yaml"
+    insert_patches_strategic_merge "$dst/kustomization.yaml" s3-migration-deployment-patch.yaml
+
+    # update the BackupstorageLocation
+    render_yaml_file "$src/tmpl-s3-migration-bsl.yaml" > "$dst/s3-migration-bsl.yaml"
+    insert_resources "$dst/kustomization.yaml" s3-migration-bsl.yaml
+
+    export WILL_MIGRATE_VELERO_OBJECT_STORE="1"
```

Comment:
what about refactoring this function into two separate functions with one `velero_should_migrate_from_object_store()`?

Reply:
Sure, pushing some changes now. Let me know if it's what you're thinking.